tests/lib/ocpp/v201/test_profile.cpp

```diff
@@ -0,0 +1,1155 @@
+    schedule1.minChargingRate = 1000.0;
+    ASSERT_NE(schedule1, schedule2);
+
+    schedule2.minChargingRate = 199.0;
+    ASSERT_NE(schedule1, schedule2);
+
+    schedule2.minChargingRate = 1000.0;
+    ASSERT_EQ(schedule1, schedule2);
+}
+
+TEST(OCPPTypesTest, CalculateChargingSchedule_Empty) {
+    std::vector<period_entry_t> combined_schedules{};
+    CompositeSchedule expected = CompositeSchedule{
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{.startPeriod = 0, .limit = NO_LIMIT_SPECIFIED}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = std::chrono::duration_cast<std::chrono::seconds>(minutes(10)).count(),
+        .scheduleStart = dt("12:00"),
+        .chargingRateUnit = ChargingRateUnitEnum::A};
+
+    CompositeSchedule actual = calculate_composite_schedule(combined_schedules, dt("12:00"), dt("12:10"), std::nullopt);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingSchedule_Exact) {
+    DateTime now = dt("12:00");
+    DateTime end = dt("12:10");
+    std::vector<period_entry_t> combined_schedules{
+        {now, end, 24.0, 3, std::nullopt, 1, ChargingRateUnitEnum::A, std::nullopt}};
+    CompositeSchedule expected = CompositeSchedule{
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{
+            .startPeriod = 0,
+            .limit = 24.0,
+            .numberPhases = 3,
+        }},
+        .evseId = EVSEID_NOT_SET,
+        .duration = std::chrono::duration_cast<std::chrono::seconds>(minutes(10)).count(),
+        .scheduleStart = now,
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+
+    CompositeSchedule actual = calculate_composite_schedule(combined_schedules, now, end, ChargingRateUnitEnum::A);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingSchedule_ShortExact) {
+    DateTime now = dt("12:00");
+    DateTime end = dt("12:10");
+    std::vector<period_entry_t> combined_schedules{{now, DateTime(end.to_time_point() - seconds(1)), 24.0, 3,
+                                                    std::nullopt, 1, ChargingRateUnitEnum::A, std::nullopt}};
+
+    CompositeSchedule expected = CompositeSchedule{
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{
+                                       .startPeriod = 0,
+                                       .limit = 24.0,
+                                       .numberPhases = 3,
+                                   },
+                                   ChargingSchedulePeriod{.startPeriod = 599, .limit = NO_LIMIT_SPECIFIED}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = std::chrono::duration_cast<std::chrono::seconds>(minutes(10)).count(),
+        .scheduleStart = now,
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+
+    CompositeSchedule actual = calculate_composite_schedule(combined_schedules, now, end, ChargingRateUnitEnum::A);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingSchedule_LongExact) {
```

Comment:
what is the distinction between short and long? Both of them have `.duration = std::chrono::duration_cast<std::chrono::seconds>(minutes(10)).count(),`

Reply:
In `CalculateChargingSchedule_ShortExact` there are two `ChargingSchedulePeriod`s. The time window is adjusted slightly in the `combined_schedules` so that you can see the different results.